tests/UtilsTests/EventTagUtilsTest.php

```diff
@@ -39,12 +46,60 @@ public function testGetRevenueValueWithInvalidRevenueTag() {
         $this->assertNull(EventTagUtils::getRevenueValue(array('revenue' => 0.5)));
         $this->assertNull(EventTagUtils::getRevenueValue(array('revenue' => '65536')));
         $this->assertNull(EventTagUtils::getRevenueValue(array('revenue' => true)));
+        $this->assertNull(EventTagUtils::getRevenueValue(array('revenue' => false)));
         $this->assertNull(EventTagUtils::getRevenueValue(array('revenue' => array(1, 2, 3))));
     }
 
     public function testGetRevenueValueWithRevenueTag() {
-        $this->assertEquals(65536, EventTagUtils::getRevenueValue(array('revenue' => 65536)));
-        $this->assertEquals(9223372036854775807, EventTagUtils::getRevenueValue(array('revenue' => 9223372036854775807)));
-        $this->assertEquals(0, EventTagUtils::getRevenueValue(array('revenue' => 0)));
+        $this->assertSame(65536, EventTagUtils::getRevenueValue(array('revenue' => 65536)));
+        $this->assertSame(9223372036854775807, EventTagUtils::getRevenueValue(array('revenue' => 9223372036854775807)));
+        $this->assertSame(0, EventTagUtils::getRevenueValue(array('revenue' => 0)));
+    }
+
+    public function testGetNumericValueInvalidArgs() {
+       $this->assertNull(EventTagUtils::getNumericValue(null));
+       $this->assertNull(EventTagUtils::getNumericValue(0.5));
+       $this->assertNull(EventTagUtils::getNumericValue(65536));
+       $this->assertNull(EventTagUtils::getNumericValue(9223372036854775807));
+       $this->assertNull(EventTagUtils::getNumericValue('65536'));
+       $this->assertNull(EventTagUtils::getNumericValue(true));
+       $this->assertNull(EventTagUtils::getNumericValue(false));
+   }
+
+   public function testGetNumericValueNoValueTag() {
+        $this->assertNull(EventTagUtils::getNumericValue(array()));
+        $this->assertNull(EventTagUtils::getNumericValue(array('non-value' => 42)));
+    }
+
+    public function testGetNumericValueWithInvalidValueTag() {
```

Comment:
all this should be non-value.

Reply:
This should use 'value'. We have already checked with the 'non-value' tags in the above function. This method is to check for invalid values passed as 'value'. I have reported the same issue in their python sdk